simple_mediation.rmd

```diff
@@ -0,0 +1,123 @@
+```{r, include=FALSE}
+options(tinytex.verbose = TRUE)
+```
+
+```{r}
+#RQ: Does domain-specifical movement behaviour profile membership mediate the relationship between socioeconomic position and socio-emotional outcomes in children?
+##Simple mediation for males
+```
+
+```{r include=FALSE}
+library(medflex)
+library(dplyr)
+library(haven)
+library(rmarkdown)
+```
+
+```{r}
+# Load male dataset from Study 2, LSAC wave 6 data, and LSAC wave 8 data
+load(file = "Z:/LSAC dataset/Study_2/Study_2/df_males_domsp_weekday.RData")
+lsac_wave6 <- read_sas("Z:/LSAC dataset/General Release/Survey data/SAS/lsacgrb10.sas7bdat")
+lsac_wave8 <- read_sas("Z:/LSAC dataset/General Release/Survey data/SAS/lsacgrb14.sas7bdat")
+```
+
+```{r}
+# Select relevant columns from male dataset (hicid, profile membership, age, remoteness, Indigenous status, SEP)
+relevant_male_data <- df_male_domsp_weekday %>%
+  select(hicid, model3_trajectory_assignments, Age, Remoteness, Indigenous, SEP)
+
+# Select relevant columns from LSAC wave 6 (SEIFA varialbes), parents living at home, parent mental health, parent ethnicity)
+wave_6_relevant_data <- lsac_wave6 %>%
+  select(hicid, fcnfsad2d, fcnfsda2d, fcnfser2d, fcnfseo2d, fmoth, ffath, fsc13a1i, zf09fm, zf09ff) 
+
+# Update column names to make sense
+colnames(wave_6_relevant_data)[2:10] <- c("adv", "dis", "resources", "education_occupation", "mother_home", "father_home", "mental_health", "mother_race", "father_race")
```

Comment:
:bulb: be a little wary of renaming this way. It relies on the columns never changing, but you can imagine a scenario where you decide you need to add a column above, but then have to remember where it was renamed.

Using tools like `rename` from {dplyr} can make this clearer, or using named vectors instead.